login/pam.c

```diff
@@ -33,11 +33,20 @@
 
 #define UNUSED(var) (void)(var)
 
-static const char *arg_value(const char *arg, const char *key) {
-  int key_len = strlen(key);
-  if (strncmp(arg, key, key_len) == 0 && arg[key_len] == '=') {
+static const char *arg_value(const char *arg, const char *key,
+                             const char *default_value) {
+  int i, key_len = strlen(key);
+  for (i = 0; i < key_len; i++) {
```

Comment:
Given that this is only bounded by `strlen(key)`, couldn't this overrun the end of the `argv` buffer?

Reply:
`arg` is a null-terminated string and if it is shorter than the `key` it will cause the if statement inside the loop to fail: null char in `arg` will not be equal to the regular char in `key`.